lib/inherited_resources.rb

```diff
@@ -27,7 +27,7 @@ def self.flash_keys=(array)
 
 ActiveSupport.on_load(:action_controller) do
   # We can remove this check and change to `on_load(:action_controller_base)` in Rails 5.2.
-  if self == ActionController::Base
+  if self == ActionController::Base || self == ActionController::API
```

Comment:
Just remove this conditional is you want to support API controllers along with the comment above.